### ISSUE TITLE
ImmutableCalendarServerAccess initialization calling setValue

### ISSUE THREAD

**Issue by xingmarc** [ical4j/ical4j-extensions#24]
Hi,

I was checking out the code and I see in [CalendarServerAccess.java](https://github.com/ical4j/ical4j-extensions/blob/develop/src/main/java/net/fortuna/ical4j/extensions/caldav/property/CalendarServerAccess.java),

When we're initializing `ImmutableCalendarServerAccess`,
it calls 
```
public CalendarServerAccess(ParameterList aList, String value) {
        super(PROPERTY_NAME, aList, new Factory());
        setValue(value);
    }
```
which eventually calls the `setValue` method defined in `ImmutableCalendarServerAccess`:
```
    private static final class ImmutableCalendarServerAccess extends CalendarServerAccess {
        private static final long serialVersionUID = -2054338254L;

        private ImmutableCalendarServerAccess(String value) {
            super(new ParameterList(true), value);
        }

        public void setValue(String aValue) {
            throw new UnsupportedOperationException("Cannot modify constant instances");
        }
    }
```

I'm wondering if I'm using this library wrong or is this a real issue.
because I see in similar class like ical4j [Method.java](https://github.com/ical4j/ical4j/blob/7ceecb058a9cde6c1bcf9ac16fdac670c4814e55/src/main/java/net/fortuna/ical4j/model/property/Method.java), the `ImmutableMethod` will call the super constructor which will use `this.value = value` and not invoking the `setValue` method.

I know that those code were there for quite a long time, so I'm wondering if I'm using this library in a wrong way.

Thanks. Looking forward to your reply.

I'm using ical4j-extension 1.0.3 and ical4j 2.2.0

**Comment by benfortuna** [comment]
Yes, I think that must be a bug. I'll put an fix in shortly.

**Comment by benfortuna** [comment]
fix applied in PR #25
